crates/burn-core/src/optim/visitor.rs

```diff
@@ -4,10 +4,11 @@ use burn_tensor::{backend::AutodiffBackend, Tensor};
 use core::marker::PhantomData;
 
 #[derive(new)]
-pub struct GradientsParamsConverter<'a, M: AutodiffModule<B>, B: AutodiffBackend> {
-    grads: B::Gradients,
-    grads_params: &'a mut GradientsParams,
+pub struct GradientsParamsConverter<'a, 'b, M: AutodiffModule<B>, B: AutodiffBackend> {
```

Comment:
I don't think we need 2 lifetimes, but I may be wrong.

Reply:
Yeah fair, don't need the flexibility.